### PR TITLE
Manually implement Promise.allSettled() within start()

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
This replaces our single use of [`Promise.allSettled()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) with a manual implementation due to browser support and reports of runtime errors.

### Why?
When we originally implemented this, I queried the intersection of browsers that support this method against our other baseline support requirements and wrongly assumed we were safe. We have been observing runtime errors from Mobile Safari 12 and Safari 12 due to their lack of support. Therefore, I thought it would be easier to do the resolution manually.

### Notes
I attempted using a proper polyfill such as https://www.npmjs.com/package/promise.allsettled however the additional library bloat wasn't worth it compared to inlining the logic. But happy to be persuaded otherwise. I will also be canarying this change to get some real-world data.

### CODE REVIEW

File: src/init.ts [hunk 1]
```diff
@@ -48,12 +48,25 @@ function startLater(
...
+    // results regardless of their resolution status. I.e. we polyfill.
+    return Promise.all(
+        activeProviders.map((provider) => provider.fetchSessionConfig())
+        .map((p) =>
+            p.then((value) => ({
+                status: "fulfilled",
+                value
+            }))
+            .catch((reason) => ({
+                status: 'rejected',
```
Comment (jakewan) [inline src/init.ts]: Does Prettier catch these single-quotes?

**Review by jakewan — APPROVED** [review]
Aside from the one formatting question, looks good to me!

File: src/init.ts (hunk 1 above)
Comment (jakewan) [reply src/init.ts]: @phamann I added a PR on the Fastly fork with a possible fix for this: https://github.com/fastly/openinsights/pull/9

Comment (phamann) [reply src/init.ts]: Thanks!
